apps/dashboard/src/components/workflow-editor/steps/delay/delay-configure.tsx

```diff
@@ -0,0 +1,137 @@
+};
+
+const delayControlsSchema = z
+  .object({
+    type: z.enum(['regular']).default('regular'),
+    amount: z.number(),
+    unit: z.nativeEnum(TimeUnitEnum),
+  })
+  .strict();
+
+export const DelayConfigure = () => {
+  const { stepSlug = '' } = useParams<{ workflowSlug: string; stepSlug: string }>();
+  const { step, refetch } = useStepEditorContext();
+  const { workflow, isReadOnly } = useWorkflowEditorContext();
+  const { uiSchema, dataSchema, values } = step?.controls ?? {};
+
+  const schema = buildDynamicZodSchema(dataSchema ?? {});
+  const newFormValues = useMemo(() => merge(buildDefaultValues(uiSchema ?? {}), values), [uiSchema, values]);
+
+  const form = useForm<z.infer<typeof delayControlsSchema>>({
+    resolver: zodResolver(schema),
+    values: newFormValues as z.infer<typeof delayControlsSchema>,
+  });
+
+  const { updateWorkflow, isPending } = useUpdateWorkflow({
+    onSuccess: () => {
+      refetch();
+      showToast({
+        children: () => (
+          <>
+            <ToastIcon variant="success" />
+            <span className="text-sm">Saved</span>
+          </>
+        ),
+        options: TOAST_CONFIG,
+      });
+    },
+    onError: () => {
+      showToast({
+        children: () => (
+          <>
+            <ToastIcon variant="error" />
+            <span className="text-sm">Failed to save</span>
+          </>
+        ),
+        options: TOAST_CONFIG,
+      });
+    },
+  });
+
+  useEffect(() => {
+    const controlErrors = flattenIssues(step?.issues?.controls);
+    Object.entries(controlErrors).forEach(([key, value]) => {
+      form.setError(key as 'amount' | 'unit', { message: value });
+    });
+  }, [step, form]);
+
+  const onSubmit = useCallback(
+    async (data: z.infer<typeof delayControlsSchema>) => {
+      if (!workflow) {
+        return false;
+      }
+
+      await updateWorkflow({
+        id: workflow._id,
+        workflow: {
+          ...workflow,
+          steps: workflow.steps.map((step) =>
+            step.slug === stepSlug ? { ...step, controlValues: { ...data } } : step
+          ),
+        },
+      });
+
+      form.reset({ ...data });
+    },
+    [workflow, form, updateWorkflow, stepSlug]
+  );
+
+  const debouncedSave = useMemo(() => debounce(onSubmit, 800), [onSubmit]);
+
+  // Cleanup debounce on unmount
+  useEffect(() => () => debouncedSave.cancel(), [debouncedSave]);
+
+  const blocker = useBlocker(() => form.formState.isDirty || isPending);
+
+  return (
+    <>
+      <SidebarContent>
+        <CommonFields />
+      </SidebarContent>
+      <Separator />
+      <SidebarContent>
+        <Form {...form}>
+          <form
+            className="flex h-full flex-col gap-2"
+            onChange={(e) => {
+              e.preventDefault();
+              e.stopPropagation();
+              debouncedSave(form.getValues());
+            }}
+          >
+            <DelayAmount dataSchema={dataSchema ?? {}} isReadOnly={isReadOnly} />
```

Comment:
Should this be made like components in InApp with the mapping and stuff? e.g.:

```jsx
  const {
    [avatarKey]: avatar,
    [subjectKey]: subject,
    [bodyKey]: body,
    [redirectKey]: redirect,
    [primaryActionKey]: primaryAction,
    [secondaryActionKey]: secondaryAction,
  } = uiSchema.properties ?? {};
  ```

Seems like an abstraction overkill for one simple input field, also how we would map these 3 ui schema components to single component - the abstraction works for the editor fields in other steps but I'm not sure about it here:

```jsx
    "uiSchema": {
                "group": "DELAY",
                "properties": {
                    "amount": {
                        "component": "DELAY_AMOUNT",
                        "placeholder": "30"
                    },
                    "unit": {
                        "component": "DELAY_UNIT",
                        "placeholder": "seconds"
                    },
                    "type": {
                        "component": "DELAY_TYPE",
                        "placeholder": null
                    }
                }
            },
  ```